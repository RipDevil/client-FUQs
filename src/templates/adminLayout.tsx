import React from 'react';
import styled from 'styled-components';
import { Row } from 'antd';

const StyledRow = styled(Row)`
  color: initial;
`;

export const AdminLayout: React.FC = ({ children }) => {
  return <StyledRow className="admin__layout">{children}</StyledRow>;
};
