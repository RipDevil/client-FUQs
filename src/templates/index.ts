export * from './layout';
export * from './admin-layout';
