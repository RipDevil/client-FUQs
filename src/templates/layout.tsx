import React from 'react';
import styled from 'styled-components';
import { Row } from 'antd';

const StyledRow = styled(Row)`
  padding-top: 40vh;
  text-align: center;
`;

export const Layout: React.FC = ({ children }) => {
  return (
    <StyledRow align="middle" justify="center">
      {children}
    </StyledRow>
  );
};
