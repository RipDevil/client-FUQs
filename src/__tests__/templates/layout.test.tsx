/* eslint-disable */
import React from 'react';
import { render } from '@testing-library/react';
import { Layout } from 'templates';

describe('Layout tests', () => {
  describe('Should render children if presented', () => {
    it('Text', () => {
      const { getByText } = render(<Layout>Test</Layout>);
      const testedElement = getByText(/Test/i);
      expect(testedElement).toBeInTheDocument();
      expect(testedElement).toMatchSnapshot();
    });

    it('Node', () => {
      const { getByTestId } = render(
        <Layout>
          <span data-testid="test">
            Text <span>Another test</span>
          </span>
        </Layout>,
      );
      const testedElement = getByTestId('test');
      expect(testedElement).toBeInTheDocument();
      expect(testedElement).toMatchSnapshot();
    });
  });

  it('Should render null if no children presented', () => {
    const { container } = render(<Layout></Layout>);
    const testedElement = container.querySelector('.ant-row');
    expect(testedElement).toBeEmpty();
    expect(testedElement).toMatchSnapshot();
  });
});
