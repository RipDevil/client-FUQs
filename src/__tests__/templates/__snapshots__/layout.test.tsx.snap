// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests Should render children if presented Node 1`] = `
<span
  data-testid="test"
>
  Text 
  <span>
    Another test
  </span>
</span>
`;

exports[`Layout tests Should render children if presented Text 1`] = `
.c0 {
  padding-top: 40vh;
  text-align: center;
}

<div
  class="ant-row ant-row-center ant-row-middle c0"
>
  Test
</div>
`;

exports[`Layout tests Should render null if no children presented 1`] = `
.c0 {
  padding-top: 40vh;
  text-align: center;
}

<div
  class="ant-row ant-row-center ant-row-middle c0"
/>
`;
